### PR TITLE
[2.0] TF 2.0 support

### PR DESCRIPTION
**Pull request by thomwolf**
Currently converted models:
- [x] BERT
- [x] GPT-2
- [x] XLNet
- [x] XLM
- [x] Transformer-XL
- [x] GPT
- [x] RoBERTa
- [x] DistilBert

With TF 2.0 Keras imperative interface and Eager, the workflow and models are suprisingly similar:
```python
import numpy
import torch
import tensorflow as tf
from pytorch_transformers import BertModel, TFBertModel, BertTokenizer

tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
pytorch_model = BertModel.from_pretrained('bert-base-uncased')
tf_model = TFBertModel.from_pretrained('bert-base-uncased')

text = "[CLS] Who was Jim Henson ? Jim [MASK] was a puppeteer [SEP]"
tokens = tokenizer.encode(text)

pytorch_inputs = torch.tensor([tokens])
tf_inputs = tf.constant([tokens])

with torch.no_grad():
    pytorch_outputs = pytorch_model(pytorch_inputs)

tf_output = tf_model(tf_inputs, training=False)

numpy.amax(numpy.abs(pytorch_outputs[0].numpy() - tf_output[0].numpy()))
# >>> 2.861023e-06 => we are good,  a few 1e-6 is the expected difference
# between TF and PT arising from internals computation ops
```

The convention is to use the same name for classes as the original PyTorch classes but prefixed with `TF`.

If you want to install and use this development branch, you should install from the `tf2` branch like this:
- install TF 2.0: `pip install tensorflow==2.0.0-rc0`
- install pytorch-transformers from the `tf2` branch: `pip install https://github.com/huggingface/pytorch-transformers/archive/tf2.zip`

TO-DO / not forget:
- [ ] check weights initialization
- [x] add weights tying
- [x] add example with losses using `model.compile` /`model.fit`
- [ ] take care of having the two possible gelu implementations for Bert
- [ ] untangle Transfo-XL tokenizer from `torch.load` and `torch.save`
- [x] test that all dropout modules are desactivated when training=False (check determinism)
- [ ] clean up our FP16 support (for PyTorch as well) with (i) an adjustment of masking values and (ii) an adjustment of LayerNorm epsilon (add an attribute in configuration files).

### CODE REVIEW

**Comment by thomwolf** [comment]
Ok big merge
